controllers/compaction/reconciler.go

```diff
@@ -195,19 +198,27 @@ func (r *Reconciler) reconcileJob(ctx context.Context, logger logr.Logger, etcd
 
 	// Delete job and requeue if the job failed
 	if job.Status.Failed > 0 {
+		if job.Status.StartTime != nil {
+			CompactionJobDurationSeconds.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededFalse}).Set(float64(time.Since(job.Status.StartTime.Time).Seconds()))
+		}
 		err = r.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground))
 		if err != nil {
 			return ctrl.Result{
 				RequeueAfter: 10 * time.Second,
 			}, fmt.Errorf("error while deleting failed compaction job: %v", err)
 		}
+		CompactionJobCounterTotal.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededFalse}).Inc()
 		return ctrl.Result{
 			RequeueAfter: 10 * time.Second,
 		}, nil
 	}
 
 	// Delete job and return if the job succeeded
 	if job.Status.Succeeded > 0 {
+		CompactionJobCounterTotal.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededTrue}).Inc()
+		if job.Status.CompletionTime != nil {
```

Comment:
Do we not also have to check for `Status.StartTime` here? it's checked above

Reply:
If the job has succeeded, then `completionTime` is sufficient, since it already accurately captures the time taken for the job to complete successfully. So I don't think we need `startTime` when the job has succeeded.

Reply:
@aaronfern is right here. `completionTime` is the time when the job completed. We are collecting the duration in the metrics. So, we need to consider `startTime`. I made necessary changes.